### ISSUE TITLE
Can habitat get the whole information of the simulation scene?

### ISSUE THREAD

**Issue by Roberyan** [facebookresearch/habitat-lab#2094]
I am working on a project to extract image data for training a model. I wonder if habitat is able to locate different object's location even in different rooms? Like I want to measure the distance of robot to any given objects in the simulation env, is this possible in habitat env? I did not see anything similar in the tutorial page.

**Comment by aclegg3** [comment]
Yes, you can get distance to objects assuming the objects are annotated (as SemanticObjects) or dynamic/kinematic (ManagedObjects).
For example, with ReplicaCAD or HSSD (dataset with inserted objects) you can get all the objects from the `RigidObjectManager` and use their `.translation` property as the input to a navmesh snap function call. That will give you a valid navigable point to reach the object (sometimes this can be more difficult to acquire). Once you have a valid nav point you can use the `ShortestPathFollower` to get a geodesic path from the agent'd position to the object position and measure it for distance.

This is actually not too expensive overall but there are several details in that exmplanation which I have't unwrapped. Feel free to ask any follow-up questions once you've looked in to the things I mentioned above or to ground the problem in your specific use case.
